lib/course_planner/courses/offered_courses.ex

```diff
@@ -79,19 +79,19 @@ defmodule CoursePlanner.Courses.OfferedCourses do
   end
 
   def create_pending_attendance_notification_map(notifiable_users) do
```

Comment:
I thought about an alternative solution using for-comprehension for this, which seems more understandable
```elixir

  def create_pending_attendance_notification_map(notifiable_users) do
    notifiable_ids = Enum.map(notifiable_users, &(&1.id))
    for oc <- with_pending_attendances(), t <- oc.teachers, t.id in notifiable_ids do
      %{
        user: t,
        type: :attendance_missing,
        path: Attendances.get_offered_course_fill_attendance_path(oc.id),
        data: %{offered_course_name: "#{oc.term.name}-#{oc.course.name}"}
      }
    end
  end
```
what do you think?

Reply:
if you like the solution you may use better names though 😅

Reply:
looks pretty cool

Reply:
i think to be honest naming is better this way. when I tried to replace t with teacher and oc with offered_course it was a bit confusing already for me to understand it

Reply:
done. replaced